python/lsst/summit/utils/astrometry/anet.py

```diff
@@ -0,0 +1,428 @@
+            ``astrometry_net_wcs_header`` : `dict`
+                The fitted wcs, as a header dict.
+        """
+        nominalWcs = exp.getWcs()
+        if nominalWcs is not None:
+            ra, dec = nominalWcs.getSkyOrigin()
+            scaleEstimate = nominalWcs.getPixelScale().asArcseconds()
+        else:
+            print('Trying to process image with None wcs - good luck!')
+            vi = exp.getInfo().getVisitInfo()
+            ra, dec = vi.boresightRaDec
+            if np.isnan(ra.asDegrees()) or np.isnan(dec.asDegrees()):
+                raise RuntimeError('Exposure has no wcs and did not find nominal ra/dec in visitInfo')
+
+        if not scaleEstimate:  # must either have a wcs or provide via kwarg
+            raise RuntimeError('Got no kwarg for scaleEstimate and failed to find one in the nominal wcs.')
+
+        image_height, image_width = exp.image.array.shape
+        scale_units = 'arcsecperpix'
+        scale_type = 'ev'  # ev means submit estimate and % error
+        scale_err = percentageScaleError  # error as percentage
+        center_ra = ra.asDegrees()
+        center_dec = dec.asDegrees()
+        radius = radius if radius else 180  # degrees
+        try:
+            wcs_header = self.adn.solve_from_source_list(sourceCat['base_SdssCentroid_x'],
+                                                         sourceCat['base_SdssCentroid_y'],
+                                                         image_width, image_height,
+                                                         scale_units=scale_units,
+                                                         scale_type=scale_type,
+                                                         scale_est=scaleEstimate,
+                                                         scale_err=scale_err,
+                                                         center_ra=center_ra,
+                                                         center_dec=center_dec,
+                                                         radius=radius,
+                                                         crpix_center=True,  # the CRPIX is always the center
+                                                         solve_timeout=240)
+        except RuntimeError:
+            print('Failed to find a solution')
+            return None
+
+        print('Finished solving!')
+
+        nominalRa, nominalDec = exp.getInfo().getVisitInfo().getBoresightRaDec()
+
+        if 'CRVAL1' not in wcs_header:
+            raise RuntimeError("Astrometric fit failed.")
+        calculatedRa = geom.Angle(wcs_header['CRVAL1'], geom.degrees)
+        calculatedDec = geom.Angle(wcs_header['CRVAL2'], geom.degrees)
+
+        deltaRa = geom.Angle(wcs_header['CRVAL1'] - nominalRa.asDegrees(), geom.degrees)
+        deltaDec = geom.Angle(wcs_header['CRVAL2'] - nominalDec.asDegrees(), geom.degrees)
+
+        # TODO: change this to return an AstrometryNetResult class like the
```

Comment:
Yeah, indeed. I'll even make a ticket for it actually (I don't always do that in this area because some things are more like notes to myself/others, rather than thing fully warranting tickets.